src/tests/resistance/fuzzing_corrupted_messages.rs

```diff
@@ -102,43 +117,51 @@ async fn corrupted_version_when_node_initiates_handshake() {
 
     let locked_rng = Arc::new(RwLock::new(seeded_rng()));
 
-    // create tcp listeners for peer set (port is only assigned on tcp bind)
-    let mut listeners = Vec::with_capacity(ITERATIONS);
+    // create peers (we need their ports to give to the node)
+    let mut peers = Vec::with_capacity(ITERATIONS);
     for _ in 0..ITERATIONS {
-        listeners.push(TcpListener::bind(new_local_addr()).await.unwrap());
+        let peer = SyntheticNode::builder()
```

Comment:
`build_n` here?

Reply:
I initially just refactored whatever there was to the builder pattern. What we can also do is move the builder outside the loop and then just call `build` inside. Alternatively we do `build_n` outside. I think I prefer just `build` for these sequential tests.

Will refactor these.